clcache.py

```diff
@@ -781,15 +788,10 @@ def expandBasedirPlaceholder(path):
 
 def collapseBasedirToPlaceholder(path):
     baseDir = normalizeBaseDir(os.environ.get('CLCACHE_BASEDIR'))
-    if baseDir is None:
-        return path
-    else:
-        assert path == os.path.normcase(path)
-        assert baseDir == os.path.normcase(baseDir)
-        if path.startswith(baseDir):
-            return path.replace(baseDir, BASEDIR_REPLACEMENT, 1)
-        else:
-            return path
+    if baseDir:
+        replacement = re.compile(re.escape(baseDir), re.IGNORECASE)
+        return replacement.sub(BASEDIR_REPLACEMENT, path)
+    return path
```

Comment:
Why would the `assert path == os.path.normcase(path)` not hold anymore? Ah, I guess it's because this function is not given paths anymore, but command line arguments, e.g. `/IC:\Foo\Bar`. That suggests that either the argument should be renamed (`path` does not fit anymore), or maybe the newly introduced `collapseBasedirInCmd` should not pass arguments to `collapseBasedirToPlaceholder` but rather the actual paths.

To extract the paths from the arguments, the code in `CommandLineAnalyzer.parseArgumentsAndFiles` could maybe get refactored to be more reusable (it already solves the issue of getting the values for command line arguments).

I guess for now, simply renaming the `path` argument to e.g. `argument` or maybe even just `s` would be fine, at least it's not misleading anymore.